web/main.ts

```diff
@@ -2490,11 +2523,23 @@ class GitGraphView {
 					});
 					const commitDetails = expandedCommit.commitDetails!;
 					const parents = commitDetails.parents.length > 0
-						? commitDetails.parents.map((parent) => {
+						? commitDetails.parents.map((parent, parentIndex) => {
 							const escapedParent = escapeHtml(parent);
-							return typeof this.commitLookup[parent] === 'number'
-								? '<span class="' + CLASS_INTERNAL_URL + '" data-type="commit" data-value="' + escapedParent + '" tabindex="-1">' + escapedParent + '</span>'
-								: escapedParent;
+
+							const parentWasLoaded = loadedParents.includes(parent);
+							const isComparedToParent = parentIndex + 1 === expandedCommit.parentIndex;
+
+							let parentHtml = abbrevCommit(escapedParent);

```

Comment:
If you're going to abbreviate each parent hash, you should do it before calling `escapeHtml` (on line 2527).